packages/paste-icons/src/react/helpers/IconWrapper.tsx

```diff
@@ -1,18 +1,35 @@
 import styled from '@emotion/styled';
+import {compose, size, system} from 'styled-system';
+import {IconSize, TextColor} from '@twilio-paste/types';
+
+// I can't name this simply `color` because TS + Emotion doesn't like it
+const iconColor = system({
+  iconColor: {
+    property: 'color',
+    scale: 'textColors',
+  },
+});
 
 export interface IconWrapperProps {
-  as?: string;
-  size?: number;
-  color?: string;
+  as?: keyof JSX.IntrinsicElements;
+  size?: IconSize;
+  iconColor?: TextColor | 'currentColor';
```

Comment:
I believe we need to add `currentColor` into our tokens list OR I can update `TextColor` in the types package maybe.  TBD